0011-R-multiple_reminders_actors_impl.md

```diff
@@ -0,0 +1,149 @@
+
+For some minor changes left in interfaces,  has been raised (https://github.com/dapr/dapr/pull/7281).
+
+## How to build and deploy Actors/Scheduler/Placement Service?
+
+There are two sub-systems that we are dealing with here: Actors and Reminders. And, as we know, going forward, there can be multiple variants of Services providing either of these sub-systems or both.
+
+There will be multiple possibilities, like:
+
+1. Placement Service used for Actors and Distributed Scheduler used for Reminders.
+
+2. ABC Service used for both Actors and Reminders.
+
+3. XYZ used for Actors and Distributed Scheduler used for Reminders.
+
+So, how to decide what all to deploy in what case for user?
+
+### Steps:
+
+1. Dapr Configuration changes to depict what services to load for Control Plane and what not.
+
+2. Component loading capability, specifically by operator, for Control Plane. This is required, in case CP requires a component to be loaded.
+
+#### Step 1: Configuration changes:
+
+A) Changes in charts/dapr/charts/dapr_config/templates/dapr_default_config.yaml
+
+B) Changes in charts/dapr/values.yaml for global values to take inputs from users.
+
+C) Changes in templates/deployment.yaml of particular service in charts to check if it needs to be deployed.
+
+##### CLI flags and annotations:
+
+* Injector is configured (via env vars set by Helm) with what actor and/or reminders service to use. Default value is to use placement and no reminders service (so it uses built-in reminders). It is possible to set in the Helm chart values to use different reminders and/or placement services.
+
+For reminders:
+
+* Default is to set nothing in the CLI, which means use the built-in reminders (that use state stores)
+* If in the Helm chart the injector is configured with a reminders service, then it adds the CLI flag `--reminders-service <name>:<address>`, for example `--reminders-service scheduler:k8s-svc-name.svc.cluster.local:port` (the `<name>` part tells daprd which implementation to load)
+
+For placement:
+
+* If the user has set the `dapr.io/placement-host-address` annotation, that forces the injector to set `--placement-host-address` to the value passed, as is the case today
+* Otherwise, if there's no explicit dapr.io/placement-host-address annotation:
+    * If the configuration (via Helm) is to use placement, then sets `--placement-host-address` exactly as today (this is default value)
+    * If the configuration is to use another actors service, it sets `--actors-service <name>:<address>:<port>` following the same logic as the reminders svc)
+This is backwards-compatible.
+
+Note: Exception is thrown when `--actors-service` or `--reminders-service` are set, if the injector is trying to inject a sidecar that use an oldeer version of Dapr, daprd that will crash because the flag doesn't exist. This is a "feature": if the cluster wants to use a different actors/reminders service, versions of Dapr that don't support that should crash.
```

Comment:
```suggestion
Note: Exception is thrown when `--actors-service` or `--reminders-service` are set, if the injector is trying to inject a sidecar that use an older version of Dapr, daprd that will crash because the flag doesn't exist. This is a "feature": if the cluster wants to use a different actors/reminders service, versions of Dapr that don't support that should crash.
```